src/Controller/Api/Produto/ConsultarController.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace App\Controller\Api\Produto;
+
+use App\Repository\ProdutoRepository;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\Routing\Attribute\Route;
+use Symfony\Component\Serializer\SerializerInterface;
+
+class ConsultarController extends AbstractController
+{
+    #[Route('/api/produto/consultar', name: 'api_produto_consultar', methods:  ['GET'])]
+    public function index(ProdutoRepository $produtoRepository, SerializerInterface $serializer): JsonResponse
+    {
+        try {
+            $produtos = $produtoRepository->findAll();
+            $json =  $serializer->serialize($produtos, 'json', ['groups' => 'produto']);
+
```

Comment:
mano, e se a gente botase uma if para se o produto nao foi encontrado
tipo assim : 
```suggestion
if(!$produtos){
    return $this->json(['message' => 'Nenhum produto encontrado'], 404);
}
``` 
mas teste ai pra ver se isso da certo kakakakaka

Reply:
Ei manin, testei aqui e deu certo, não tinha me atentado a essa possibilidade, obrigado